File: src/main/java/com/kishlaly/ta/analyze/tasks/blocks/groups/BullishDivergence_Buy_1.java
package com.kishlaly.ta.analyze.tasks.blocks.groups;

import com.kishlaly.ta.analyze.tasks.blocks.TaskBlockJava;
import com.kishlaly.ta.analyze.tasks.blocks.commons.ScreenBasicValidationJava;
import com.kishlaly.ta.analyze.tasks.blocks.one.Long_ScreenOne_StrictTrendCheckJava;
import com.kishlaly.ta.analyze.tasks.blocks.two.Long_ScreenTwo_BullishDivergenceMainLogicJava;

import java.util.ArrayList;
import java.util.List;

import static com.kishlaly.ta.analyze.tasks.DivergenciesJava.BullishConfig.ALLOW_ON_BEARISH_TREND;

public class BullishDivergence_Buy_1 implements BlocksGroupJava {

    public List<TaskBlockJava> blocks() {
        return new ArrayList<TaskBlockJava>() {{
            add(new ScreenBasicValidationJava());
            if (!ALLOW_ON_BEARISH_TREND) {
                add(new Long_ScreenOne_StrictTrendCheckJava());
            }

            add(new Long_ScreenTwo_BullishDivergenceMainLogicJava());
        }};
    }

    @Override
    public String comments() {
        return "Not stable. Pls fix me";
    }

}

File: src/main/kotlin/com/kishlaly/ta/analyze/tasks/blocks/groups/BullishDivergence_Buy_1.kt
package com.kishlaly.ta.analyze.tasks.blocks.groups

import com.kishlaly.ta.analyze.tasks.Divergencies
import com.kishlaly.ta.analyze.tasks.blocks.TaskBlock
import com.kishlaly.ta.analyze.tasks.blocks.commons.ScreenBasicValidation
import com.kishlaly.ta.analyze.tasks.blocks.one.Long_ScreenOne_StrictTrendCheck
import com.kishlaly.ta.analyze.tasks.blocks.two.complex.Long_ScreenTwo_BullishDivergenceMainLogic

class BullishDivergence_Buy_1 : BlocksGroup {

    override fun blocks(): List<TaskBlock> {
        return object : ArrayList<TaskBlock>() {
            init {
                add(ScreenBasicValidation())
                if (!Divergencies.BullishConfig.ALLOW_ON_BEARISH_TREND) {
                    add(Long_ScreenOne_StrictTrendCheck())
                }

                add(Long_ScreenTwo_BullishDivergenceMainLogic())
            }
        }
    }

    override fun comments() = "Not stable. Pls fix me"

}